fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -26,6 +26,42 @@
 
 namespace {
 
+class RocksDBErrorListener : public rocksdb::EventListener {
+public:
+	RocksDBErrorListener(ThreadReturnPromise<Void>* promise) : errorPromise(promise){};
+	void OnBackgroundError(rocksdb::BackgroundErrorReason reason, rocksdb::Status* bg_error) override {
+		TraceEvent(SevError, "RocksDBBGError")
+		    .detail("Reason", reason)
+		    .detail("RocksDBSeverity", bg_error->severity())
+		    .detail("Status", bg_error->ToString());
+		std::unique_lock<std::mutex> lock(mutex);
+		if (isSet)
```

Comment:
This seems duplicative of `Promise::isSet()`. We should just expose that in the `ThreadReturnPromise` API instead of implementing our own version.

Reply:
Done